app/models/hackney/repair_request.rb

```diff
@@ -63,4 +71,36 @@ def self.attributes_from_api(a)
       property_reference: a['propertyReference']
     }
   end
+
+  private
+
+  # TODO: API should give better error descriptions
+  def add_nested_errors
+    errors.each do |key, msg|
+      case key
+      when /^contact\.(.*)$/i
+        contact.errors.add($1, msg)
+      when /^work_orders\[0\]\.(.*)$/i
+        work_orders[0].errors.add($1, msg)
+      end
+    end
+  end
+
+  # TODO: API should give better error descriptions
+  def self.errors_from_api(e)
+    e.map {|x| x["userMessage"] }.group_by do |x|
```

Comment:
seems weird that we make strings and then need to parse them with regexes later on

Reply:
indeed this stinks. I just had an idea now that could solve this. Just a sec.

Reply:
would it work better as an instance method? Then you could return pairs: `[self.contact, :telephone_number], [self, :description], [work_orders[0], :sor_code]`